lib/crystalball/map_generator/allocated_objects_strategy.rb

```diff
@@ -35,6 +35,8 @@ def call(case_map)
         objects = object_tracker.created_during do
           yield case_map
         end
+        described_class = case_map.example.metadata[:described_class]
+        objects.push(case_map.example.metadata[:described_class]) if described_class.is_a?(Class)
```

Comment:
that conflicts with strategy purpose. object wasn't allocated during an example. Why do you this?

Reply:
That test is senseless. Are you testing Ruby there?

In theory there might be a case when you call pure class level DSL inside of an extended module, however pushing `described_class` doesn't save you from this flaw since it's just a workaround for one specific case (when your extended class is described_class). 
`UsedObjectsStrategy` is also poor name because it doesn't return all used objects. E.g.
```
class A; end
class B; end
describe A do
 specify { expect(described_class.ancestors.include?(B)).to eq false }
end
```
B is used but won't be in a map. So I prefer a strategy to do fully what it claims to do rather than a strategy doing a bit more with side effects.

Reply:
You can consider creating a separate strategy that will be a replacement for prediction strategy `AssociatedSpecs.new from: 'model.rb', to: 'model_spec.rb'`.

Something that reuses the same DefinitionTracer and adds `described_class` definition_paths to the map. That will cover slightly more than current prediction strategy because it will handle cases where a class spec is distributed across several files\contexts.